### ISSUE TITLE
Upgrade to Rails 7.0.8

### ISSUE THREAD

**Issue by ryan-mcneil** [department-of-veterans-affairs/va.gov-team#69300]
### Description
Our "end goal" here is to get Rails to the latest version. To properly do that, we should do it in a few steps, as outlined under Tasks. This is blocked by 1 or more gem upgrades, as linked to this ticket.

### Tasks
- [x] Update to ~> 7.0 (7.0.8)
- [x] Fix any broken tests (currently seeing 200+ in this PR)
- [ ] Merge to master and test in staging
- [ ] Standby for Prod deploy

### Acceptance Criteria
- [ ] vets-api + modules are all at the latest Rails version in all environments

**Comment by ryan-mcneil** [comment]
Update: [New PR started](https://github.com/department-of-veterans-affairs/vets-api/pull/14904). Working through how we use sessions in the test env. This seems to be causing the majority of the failing tests.
